pkg/snapstore/snapstore_test.go

```diff
@@ -278,6 +280,220 @@ var _ = Describe("Save, List, Fetch, Delete from mock snapstore", func() {
+	},
+	// Swift JSON
+	{
+		Provider:          "Swift",
+		EnvVariable:       "OPENSTACK_APPLICATION_CREDENTIALS_JSON",
+		SnapstoreProvider: brtypes.SnapstoreProviderSwift,
+		CredentialType:    "file",
+		CredentialFiles:   []string{"credentials.json"},
+	},
+	// OSS
+	{
+		Provider:          "OSS",
+		EnvVariable:       "ALICLOUD_APPLICATION_CREDENTIALS",
+		SnapstoreProvider: brtypes.SnapstoreProviderOSS,
+		CredentialType:    "directory",
+		CredentialFiles:   []string{"accessKeyID", "accessKeySecret", "storageEndpoint"},
+	},
+	{
+		Provider:          "OSS",
+		EnvVariable:       "ALICLOUD_APPLICATION_CREDENTIALS_JSON",
+		SnapstoreProvider: brtypes.SnapstoreProviderOSS,
+		CredentialType:    "file",
+		CredentialFiles:   []string{"credentials.json"},
+	},
+	// OCS
+	{
+		Provider:          "OCS",
+		EnvVariable:       "OPENSHIFT_APPLICATION_CREDENTIALS",
+		SnapstoreProvider: brtypes.SnapstoreProviderOCS,
+		CredentialType:    "directory",
+		CredentialFiles:   []string{"accessKeyID", "region", "endpoint", "secretAccessKey"},
+	},
+	{
+		Provider:          "OCS",
+		EnvVariable:       "OPENSHIFT_APPLICATION_CREDENTIALS_JSON",
+		SnapstoreProvider: brtypes.SnapstoreProviderOCS,
+		CredentialType:    "file",
+		CredentialFiles:   []string{"credentials.json"},
+	},
+}
+
+var _ = Describe("Dynamic access credential rotation test for each provider", func() {
+	for _, config := range credentialTestConfigs {
+		config := config
+		Context(fmt.Sprintf("testing secret modification for %s with %s", config.Provider, config.EnvVariable), func() {
+			Context("environment variable not set", func() {
+				It("should return error", func() {
+					newSecretModifiedTime, err := GetSnapstoreSecretModifiedTime(config.SnapstoreProvider)
+					Expect(err).Should(HaveOccurred())
+					Expect(newSecretModifiedTime.IsZero()).Should(BeTrue())
+				})
+			})
+			Context("environment variable set", func() {
+				var credentialDirectory string
+				BeforeEach(func() {
+					credentialDirectory = GinkgoT().TempDir()
+					GinkgoT().Setenv(config.EnvVariable, credentialDirectory)
+					// config.CredentialType == "file" -> env variable set to file
+					if config.CredentialType == "file" {
+						GinkgoT().Setenv(config.EnvVariable, filepath.Join(credentialDirectory, config.CredentialFiles[0]))
+					}
+				})
+				Context("credentials do not exist", func() {
+					// all files are missing
+					It("should return error when all files are missing", func() {
+						newSecretModifiedTime, err := GetSnapstoreSecretModifiedTime(config.SnapstoreProvider)
+						Expect(err).Should(HaveOccurred())
+						Expect(newSecretModifiedTime.IsZero()).Should(BeTrue())
+					})
+					// one file is missing
+					for _, credentialFile := range config.CredentialFiles {
+						credentialFile := credentialFile
+						It(fmt.Sprintf("should return error when the file '%s' is missing", credentialFile), func() {
```

Comment:
```suggestion
						It(fmt.Sprintf("should return error when the file %q is missing", credentialFile), func() {
```

Reply:
Addressed in [8635c10](https://github.com/gardener/etcd-backup-restore/pull/670/commits/8635c101d031f890343da3177cb410aa20053dc5)